bridge_adaptivity/module/templates/module/collection_form.html

```diff
@@ -16,7 +16,7 @@ <h2>Edit collection: {{collection.name }}</h2>
   {% bootstrap_form form %}
   <button class="btn btn-primary" type="submit">
     {% if collection.id %}Update{% else %}Create{% endif %}</button>
-  <a href="{% if collection.id %}{% url 'module:collection-detail' collection.id %}{% else %}{% url 'module:collection-list' %}{% endif %}" class="btn btn-default" role="button"
+  <a href="{% if back_url or not collection.id %}{% url 'module:collection-list' %}{% else %}{% url 'module:collection-detail' collection.id %}{% endif %}" class="btn btn-default" role="button"
```

Comment:
probably it should be better to move this logic to View class ?

Reply:
Agree.